src/hooks/chatting/useChatMessages.ts

```
@@ -14,47 +14,108 @@ const useChatMessages = (props: useChatMessagesProps) => {
+
+		latestPageMessageIdRef.current = latestPageMessageId;
+		processedPageCountRef.current = messagePages.length;
+
+		if (!isLatestPageUpdated && !hasNewPage) return;
+
 		setChatHistory((prevChatHistory) => {
-			const lastPageMessages = messagePages[messagePages.length - 1]?.chatChannelMessages ?? [];
+			const previousMessages = prevChatHistory?.chatChannelMessages ?? [];
+			const messagesToMerge: Message[] = isLatestPageUpdated
+				? latestPageMessages
+				: messagePages[messagePages.length - 1]?.chatChannelMessages ?? [];
```

ESLint

[error] Replace `messagePages[messagePages.length·-·1]?.chatChannelMessages·??·[]` with `(messagePages[messagePages.length·-·1]?.chatChannelMessages·??·[])`

(prettier/prettier)